### PR TITLE
adds a se layer and tests

### PR DESCRIPTION
**Pull request by wyli**
adds a channel squeeze and excitation layer
a building block for #91 #396 

### Status
**Ready**

### Types of changes

- [x] non-breaking new feature
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: monai/networks/blocks/squeeze_and_excitation.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import torch.nn as nn
+
+from monai.networks.layers.factories import Pool
+
+SUPPORTED_ACTI_1 = {"relu": nn.ReLU, "relu6": nn.ReLU6, "leakyrelu": nn.LeakyReLU}
+
+SUPPORTED_ACTI_2 = {
+    "relu": nn.ReLU,
+    "prelu": nn.PReLU,
+    "relu6": nn.ReLU6,
```
Comment (Nic-Ma) [inline monai/networks/blocks/squeeze_and_excitation.py:21]: I suggest adding `ReLU6` and `Sigmoid`, `Softmax` to our `Act factory`, @ericspod what do you think?
Thanks.

Comment (ericspod) [reply monai/networks/blocks/squeeze_and_excitation.py:21]: Yes we can add the following to `monai/networks/layers/factories.py` and include a few other activation functions provided by Pytorch:

```python
Act.add_factory_callable("relu6", lambda: nn.modules.ReLU6)
Act.add_factory_callable("selu", lambda: nn.modules.SELU)
Act.add_factory_callable("celu", lambda: nn.modules.CELU)
Act.add_factory_callable("gelu", lambda: nn.modules.GELU)
Act.add_factory_callable("sigmoid", lambda: nn.modules.Sigmoid)
Act.add_factory_callable("tanh", lambda: nn.modules.Tanh)
Act.add_factory_callable("softmax", lambda: nn.modules.SoftMax)
Act.add_factory_callable("logsoftmax", lambda: nn.modules.LogSoftMax)
```

The arguments for choosing activation would then be members of `Act`, eg. `acti_type_1: str = Act.RELU`. We could add a check in the constructor to enforce the constraint of which activations to use but I'd leave it open if people wanted to try activations not mentioned in the paper.

File: monai/networks/blocks/squeeze_and_excitation.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        )
+
+    def forward(self, x):
+        b, c = x.shape[:2]
+        y = self.avg_pool(x).view(b, c)
+        y = self.fc(y).view([b, c] + [1] * (x.ndim - 2))
+        return x * y
+
+
+class ResidualSELayer(nn.Module):
```
Comment (ericspod) [inline monai/networks/blocks/squeeze_and_excitation.py:70]: Why not inherit from ChannelSELayer?

```python
class ResidualSELayer(ChannelSELayer):
    """
    A "squeeze-and-excitation"-like layer with a residual connection.
    See also ::py:class:`monai.networks.blocks.ChannelSELayer`.
    """
    def forward(self, x):
        return x + super().forward(x)
```
